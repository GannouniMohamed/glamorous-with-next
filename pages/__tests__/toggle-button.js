import React from 'react'
import renderer from 'react-test-renderer'
import {render} from 'enzyme'
import snapshotDiff from 'snapshot-diff'
import glamorSerializer from 'jest-glamor-react'
import prettyFormat from 'pretty-format'
import ToggleButton from '../toggle-button'

test('has off styles applied by default', () => {
  const wrapper = render(<ToggleButton />)
  expect(wrapper).toMatchSnapshot()
})

test('has different styles when on: true', () => {
  expect(
    snapshotDiff(serialize(<ToggleButton />), serialize(<ToggleButton on />))
  ).toMatchSnapshot()
})

function serialize(ui) {
  return glamorSerializer
    .print(renderer.create(ui).toJSON(), val =>
      prettyFormat(val, {
        plugins: [prettyFormat.plugins.ReactTestComponent],
      })
    )
    .trim()
}
