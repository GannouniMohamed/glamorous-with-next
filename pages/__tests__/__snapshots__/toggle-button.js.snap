// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has different styles when on: true 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -8,22 +8,22 @@
   line-height: 1.4;
   text-align: center;
   cursor: pointer;
   border-radius: 4px;
   color: #fff;
-  background-color: #337ab7;
-  border-color: #285f8f;
+  background-color: #22527b;
+  border-color: #173853;
 }
 
 .glamor-0:hover,
 [data-glamor-0]:hover,
 .glamor-0:active,
 [data-glamor-0]:active,
 .glamor-0:focus,
 [data-glamor-0]:focus {
-  background-color: #285f8f;
-  border-color: #1d4567;
+  background-color: #1d4567;
+  border-color: #122a3f;
 }
 
 <button
   className="glamor-0"
 />
`;

exports[`has off styles applied by default 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline-block;
  padding: 6px 12px;
  margin-bottom: 0px;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.4;
  text-align: center;
  cursor: pointer;
  border-radius: 4px;
  color: #fff;
  background-color: #337ab7;
  border-color: #285f8f;
}

.glamor-0:hover,
[data-glamor-0]:hover,
.glamor-0:active,
[data-glamor-0]:active,
.glamor-0:focus,
[data-glamor-0]:focus {
  background-color: #285f8f;
  border-color: #1d4567;
}

<button
  class="glamor-0"
/>
`;
